src/DriverFactory.php

```diff
@@ -123,4 +129,32 @@ private function registerAnnotationLoader(): void
 
         self::$isAnnotationLoaderRegistered = true;
     }
+
+    /**
+     * @param array<string, mixed> $config
+     */
+    private function createCachedReader(ContainerInterface $container, array $config, Reader $reader): PsrCachedReader
+    {
+        $cache = $this->retrieveDependency(
+            $container,
+            $config['cache'],
+            'cache',
+            CacheFactory::class
+        );
+
+        if ($cache instanceof Cache) {
+            $cache = CacheAdapter::wrap($cache);
+        }
+
+        if ($cache instanceof CacheItemPoolInterface) {
+            return new PsrCachedReader($reader, $cache);
+        }
+
+        $cacheType = is_object($cache) ? get_class($cache) : gettype($cache);
+        assert($cacheType !== '');
+
+        throw InvalidArgumentException::fromUnsupportedCacheType(
```

Comment:
This is actually new. Before this patch, we only got `TypeError` as the return type of `CacheFactory` is never validated.

Not sure why psalm is not yelling at it but it can return every type and we only see problems when it is passed to the `__construct` of the reader.